### PR TITLE
Add special handling for multi:softmax in sklearn predict

### LINKED ISSUE #7605: Dramatic XGBoost accuracy degradation on Iris dataset.
Hi,
I have measured the XGBoost accuracy on the Iris dataset and have found dramatic degradation while switching from branch release_1.5.1 to master.

I have built xgboost from sources according to the instructions for Linux.
You can find a minimal code for reproducing the problem below:

```
import xgboost
import sklearn.datasets
import sklearn.model_selection
import sklearn.metrics

X, y = sklearn.datasets.load_iris(return_X_y=True)
Xtrn, Xtst, ytrn, ytst = sklearn.model_selection.train_test_split(
    X, y, train_size=0.8, random_state=4)

clf = xgboost.XGBClassifier(
    max_depth=3,
    learning_rate=1,
    n_estimators=10,
    silent=True,
    objective='multi:softmax',
    seed=5)
clf.fit(Xtrn, ytrn)
ypred = clf.predict(Xtst)
acc = sklearn.metrics.accuracy_score(ytst, ypred)

print('xgboost accuracy on iris:', acc)

```

If the branch "release_1.5.1" is in use, the xgboost accuracy is >0.96, but for the master branch accuracy is 0.7

### PR DESCRIPTION
**Pull request by hcho3**
Closes #7605

@trivialfis

### CODE REVIEW

File: python-package/xgboost/sklearn.py [hunk 1]
```diff
@@ -1419,6 +1419,8 @@ def predict(
             # multi-label
             column_indexes = np.zeros(class_probs.shape)
             column_indexes[class_probs > 0.5] = 1
+        elif self.objective == "multi:softmax":
+            return class_probs.astype(np.int32)
         else:
             # turns soft logit into class label
```
Comment (hcho3) [inline python-package/xgboost/sklearn.py:1425]: This general logic does not work with `multi:softmax`.

**Comment by hcho3** [comment]
@trivialfis Done.

**Review by trivialfis — APPROVED** [review]
